ogre2/src/Ogre2Material.cc

```diff
@@ -636,16 +637,82 @@ void Ogre2Material::SetTextureMapImpl(const std::string &_texture,
     }
   }
 
+  Ogre::Root *root = Ogre2RenderEngine::Instance()->OgreRoot();
+  Ogre::TextureGpuManager *textureMgr =
+      root->getRenderSystem()->getTextureGpuManager();
+
+  // workaround for grayscale emissive texture
+  // convert to RGB otherwise the emissive map is rendered red
+  if (_type == Ogre::PBSM_EMISSIVE &&
+      !this->ogreDatablock->getUseEmissiveAsLightmap())
+  {
+    common::Image img(_texture);
+    // check for 8 bit pixel
+    if (img.BPP() == 8u)
+    {
+      std::string p = common::parentPath(_texture);
```

Comment:
```suggestion
      std::string parentDirectory = common::parentPath(_texture);
```
more descriptive name?

Reply:
a6796ac